technologies/convince/README.md

```diff
@@ -3,14 +3,248 @@
+Result: 0
+```
+
+This means that the property does not hold.
+This means our alarm also does not work as expected.
+
+### Task 1.3: Make sure that the battery can't drain below 0
+
+To fix the model, you need to change the `battery_drainer.scxml` file.
+
+In lines 22 to 24, you can see:
+
+```xml
+<ros_rate_callback name="my_timer" target="use_battery">
+    <assign location="battery_percent" expr="battery_percent - 1" />
+</ros_rate_callback>
+```
+
+This gets is a transitions that gets triggered every time the timer `my_timer` fires.
+Because it's target is `use_battery`, the state will not change.
+But as you can see, the battery level is decreased by 1 every time the timer fires.
+We need to make sure that the transition is not taken when the battery level is already 0.
+
+To do this, we can add a condition to the transition:
+Add the attribute `cond="battery_percent > 0"` to the `<ros_rate_callback>` tag.
+
+Now run `as2fm_scxml_to_jani main.xml` and `smc_storm --model main.jani --properties-names battery_over_depleted` again.
+
+The output should now be:
+
+```txt
+CONVINCE Statistical Model Checker
+Checking model: main.jani
+Property "battery_over_depleted": Pmin=? [F ((topic_level_msg.ros_fields__data <= -1) & topic_level_msg.valid)];
+Result: 0
+```
+
+This means that the property does not hold anymore, which is what we expected: The battery level should never reach -1 or lower.
+
+But this has not fixed the alarm yet. Feel free to running the model checker with the `alarm_on` property again. It should still output 0.
+
+### Task 1.4: Enable the alarm
+
+To enable the alarm, we need to make sure that the `battery_manager` is evaluating the battery level correctly.
+Please convince yourself that the battery level is correctly sent.
+This is done in lines 18 to 20 of the `battery_drainer.scxml` file:
+
+```xml
+<ros_topic_publish name="battery_level">
+    <field name="data" expr="battery_percent" />
+</ros_topic_publish>
+```
+
+It is received in the `battery_manager` in lines 22 to 24 of the `battery_manager.scxml` file:
+
+```xml
+<ros_topic_callback name="level" target="check_battery">
+    <assign location="battery_alarm" expr="false" />
+</ros_topic_callback>
+```
+
+This is again a transition that is triggered every time the topic `level` is received.
+And it also does not change the state, because the target is `check_battery`.
+But it assigns the value of `false` to the variable `battery_alarm`.
+This is not what we want.
+We want to set the alarm to `true` when the battery level is below 30.
+So please change the `expr` attribute to `_msg.data &lt; 30`.
+Note that we need to use `&lt;` instead of `<` because of XML syntax.
+Note also that `_msg` is a variable that is automatically created to hold the content of the received ROS message.
+And the field `data` is the field of the `std_msgs/Int32` message that we are sending.
```

Comment:
Suggest making these 2 notes a separate paragraph and bulleted list. This whole paragraph will look long when rendered.